<a href="https://colab.research.google.com/github/mflagosv25/Talento-Tech/blob/main/Ejercicios%20y%20Tareas/Tarea_Metricas_Clasificacion.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import numpy as np
from sklearn.metrics import classification_report

In [ ]:
#Matriz de confusión de 2X2
mc = np.array([[35,3],[2,56]])
print('Matriz de confución:\n',mc)
# Genera el informe de clasificación
y_true = []
y_pred = []
for i in range(mc.shape[0]):
  for j in range(mc.shape[1]):
    for k in range(mc[i, j]):
      y_true.append(i)
      y_pred.append(j)
report = classification_report(y_true, y_pred)
# Imprime el informe
print(report)

Matriz de confución:
 [[35  3]
 [ 2 56]]
              precision    recall  f1-score   support

           0       0.95      0.92      0.93        38
           1       0.95      0.97      0.96        58

    accuracy                           0.95        96
   macro avg       0.95      0.94      0.95        96
weighted avg       0.95      0.95      0.95        96



In [ ]:
#Cálculo de métricas a partir de la matriz de confución
vn= mc[0,0]
fp = mc[0,1]
fn = mc[1,0]
vp = mc[1,1]
#Precisión = Verdaderos Positivos / (Verdaderos Positivos + Falsos Positivos)
precision = vp/(vp+fp)
#Exactitud = (Verdaderos Positivos + Verdaderos Negativos) / Total
accuracy = (vp+vn)/(vn+fp+fn+vp)
#Sensibilidad(Recall) = Verdaderos Positivos / (Verdaderos Positivos + Falsos Negativos)
recall = vp/(vp+fn)
#Recuperación = Verdaderos Positivos / (Verdaderos Positivos + Falsos Negativos)
recovery = vp/(vp+fn)
#F1-score = 2 * (Precisión * Recuperación) / (Precisión + Recuperación)
f1_score = 2*(precision*recovery)/(precision+recovery)
print('precision: ',precision.round(2))
print('recall: ',recall.round(2))
print('f1-score: ',f1_score.round(2))
print('accuracy: ',accuracy.round(2))


precision:  0.95
recall:  0.97
f1-score:  0.96
accuracy:  0.95


              precision    recall  f1-score   support

           0       0.95      0.92      0.93        38
           1       0.95      0.97      0.96        58

    accuracy                           0.95        96
   macro avg       0.95      0.94      0.95        96
weighted avg       0.95      0.95      0.95        96
